cdap-app-fabric/src/test/java/co/cask/cdap/gateway/handlers/OperationsDashboardHttpHandlerTest.java

```diff
@@ -16,105 +16,257 @@
+    } else if (statusTime.getProgramStatus().equals(ProgramRunStatus.RUNNING)) {
+      runRecordBuilder.setRunTime(statusTime.getTime());
+    } else if (statusTime.getProgramStatus().equals(ProgramRunStatus.SUSPENDED)) {
+      runRecordBuilder.setSuspendTime(statusTime.getTime());
+    } else if (statusTime.getProgramStatus().equals(ProgramRunStatus.RESUMING)) {
+      runRecordBuilder.setResumeTime(statusTime.getTime());
+    }
+    runRecordBuilder.setStatus(statusTime.getProgramStatus());
+    runRecordBuilder.setSourceId(SOURCE_ID);
+    return runRecordBuilder.build();
+  }
+
+  @Test
+  public void testDashboardWithNamespaceFiltering() throws Exception {
+    long startTime1 = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());
+    RunId runId1 = RunIds.generate();
+
+    RunRecordMeta meta =
+      getMockRunRecordMeta(NamespaceId.DEFAULT,
+                           runId1, new StatusTime(ProgramRunStatus.RUNNING, startTime1), null);
+    writeRunRecordMeta(meta, startTime1);
+
+    // write heart beat messages for 10 minutes (every minute) for this program run.
+    long endTime = startTime1 + TimeUnit.MINUTES.toSeconds(10);
+    long interval = TimeUnit.MINUTES.toSeconds(1);
+    setUpProgramHeartBeats(meta, startTime1, endTime, interval);
+
+    // write end notification message
+    meta =
+      getMockRunRecordMeta(NamespaceId.DEFAULT, runId1, new StatusTime(ProgramRunStatus.COMPLETED, endTime), null);
+    writeRunRecordMeta(meta, endTime);
+
+
+    long startTime2 = startTime1 + TimeUnit.MINUTES.toSeconds(5);
+    NamespaceId ns1 =  new NamespaceId("ns1");
+    RunId runId2 = RunIds.generate();
+    meta =
+      getMockRunRecordMeta(ns1, runId2, new StatusTime(ProgramRunStatus.RUNNING, startTime2), null);
+
+    writeRunRecordMeta(meta, startTime2);
+
+    // write heart beat messages for 5 minutes (every minute) for this program run.
+    // this program doesnt have endTime for this testOp
+    setUpProgramHeartBeats(meta, startTime2, endTime, interval);
+
+    String opsDashboardQueryPath =
+      String.format("%s/dashboard?start=%s&duration=%s&namespace=%s&namespace=%s", BASE_PATH,
+                    String.valueOf(startTime1), String.valueOf(endTime),
+                    NamespaceId.DEFAULT.getNamespace(), ns1.getNamespace());
+    // get ops dashboard query results
+    HttpResponse response = doGet(opsDashboardQueryPath);
+    Assert.assertEquals(200, response.getStatusLine().getStatusCode());
+    String content = new String(ByteStreams.toByteArray(response.getEntity().getContent()), Charsets.UTF_8);
+    List<DashboardProgramRunRecord> dashboardDetail = GSON.fromJson(content, DASHBOARD_DETAIL_TYPE);
+    // assert the result contains 2 entries
+    Assert.assertEquals(2, dashboardDetail.size());
+    Iterator<DashboardProgramRunRecord> dashboardProgramRunRecordIterator = dashboardDetail.iterator();
+
+    while (dashboardProgramRunRecordIterator.hasNext()) {
```

Comment:
this way of comparing seems awkward. Is there a reason we can't create the expected objects and compare sets? Seems like we have all the required information.